prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -1815,18 +1815,36 @@ class Hl7Serializer(
         receivingApplicationReportIn: String? = null,
         receivingFacilityReportIn: String? = null
     ): String {
-        val sendingApplicationReport = sendingApplicationReportIn
+        val hl7Config = report.destination?.translation as? Hl7Configuration?
+        val replaceValueAwithB = hl7Config?.replaceValueAwithB ?: emptyMap()
+        var sendingApplicationReportInReplace: String? = null
+        var receivingApplicationReportInReplace: String? = null
+        var receivingFacilityReportInReplace: String? = null
+        replaceValueAwithB.forEach { segment ->
```

Comment:
Should add a comment here that explains this will change both the FHS and BHS field and will only work for FHS-3,5&6. Something like. "Modifies batch headers to defined value in receiver setting replaceValueAwithB. Will only work for fields FHS-3, FHS-5 or FHS-6". Value will also be applied to BHS-3, 5 or 6 accordingly.